### PR TITLE
Feature/stage renaming and moving

### PR DESCRIPTION
**Pull request by ollikehy**
Tyyliä vois vähän parannella jos joku keksii miten

### CODE REVIEW

File: client/src/redux/sagas/jobPostingSaga.js [hunk 1]
```diff
@@ -69,6 +69,10 @@ function* removeStageInJobPosting({ payload }) {
   yield put(actions.removeStageInJobPosting(payload.stage))
 }
 
+function* renamePostingStage( { payload }) {
```
Comment (llmlks) [inline client/src/redux/sagas/jobPostingSaga.js]: This saga is not needed, as there are no side effects to the action

**Comment by llmlks** [comment]
On functionality & look:

When user clicks on a stage to be edited, the focus doesn't automatically go to the input field. Instead, the user has to click the field a second time before starting to type

User has to press enter to keep the change, but clicking outside the field doesn't hide it either, even when clicking on another stage. This is not very intuitive

Could the problem with the length of the input field be solved by using a normal input field and removing the chip, when user clicks on it? The input field could look similar to the one for title on the same page. Then there could be two buttons, one for cancelling (a red cross) and one for accepting the change made (a green tick), next to the input field. 

Is there a reason the user shouldn't be able to rename the default stages?

File: client/src/components/jobPosting/JobPostingStages.js [hunk 2]
```diff
@@ -75,35 +102,43 @@ export class JobPostingStages extends Component {
...
+                    id={jobPostingStage.stageName + '_chip'}
+                    label={index + 1 + '. ' + jobPostingStage.stageName}
+                    onClick={() => this.handleStageClick(jobPostingStage)}
+                    onDelete={() => this.handleStageDelete(jobPostingStage)} />
+                  <form id={jobPostingStage.stageName} onChange={this.handleStageRenameChange} onSubmit={(e) => this.handleStageRename(e, jobPostingStage)} className={classNames + '__stage-name-edit'}>
+                    <input id={`${jobPostingStage.stageName}_input`} onBlur={() => this.handleStageHide(jobPostingStage)} className={classNames + '__stage-name-edit__input'} type="text"></input>
+                  </form>
+                </div>
               }
               {jobPostingStage.canRemove === false &&
```
Comment (llmlks) [inline client/src/components/jobPosting/JobPostingStages.js]: The name of a stage should be editable even when it has applicants in it

File: client/src/components/jobPosting/JobPostingStages.js [hunk 3]
```diff
@@ -4,22 +4,28 @@ import { TextField, Button, Chip } from '@material-ui/core'
...
+      stageUnderEdit: '',
+      helperText: props.helperText
     }
   }
 
   componentDidMount() {
     if (this.props.stages !== null) {
       this.addCopiedStagesToNewJobPosting()
     }
+    const stageNames = this.props.jobPostingStages.map((stage) => { return stage.stageName })
```
Comment (josalmi) [inline client/src/components/jobPosting/JobPostingStages.js]: no need for return if curly brackets are removed

File: client/src/components/jobPosting/JobPostingStages.js [hunk 4]
```diff
@@ -67,6 +67,53 @@ export class JobPostingStages extends Component {
...
+    this.setState({
+      stageUnderEdit: e.target.value
+    })
+  }
+
+  handleStageRename = (e, jobPostingStage) => {
+    e.preventDefault()
+    if (this.verifyStageName(this.state.stageUnderEdit)) {
+      this.props.renamePostingStage(jobPostingStage, this.state.stageUnderEdit)
+      document.getElementById(jobPostingStage.stageName).style.display = 'none'
```
Comment (josalmi) [inline client/src/components/jobPosting/JobPostingStages.js]: No dom manipulation if possible to some other way

File: client/src/components/jobPosting/JobPostingStages.js [hunk 5]
```diff
@@ -75,41 +122,64 @@ export class JobPostingStages extends Component {
...
+        <div className={classNames + '__new-stage-add-button'}>
           <Button mini variant="fab" color="inherit" aria-label="Add" onClick={() => this.addNewStage()}>
             <AddIcon />
           </Button>
         </div>
-        <div className='job-posting-form-stages__stage-list'>
+        <div className={classNames + '__stage-list'}>
           {this.props.jobPostingStages.map((jobPostingStage, index) => (
-            <div key={index} className='job-posting-form-stages__single-stage'>
+            <div key={index} className={classNames + '__single-stage'}>
```
Comment (josalmi) [inline client/src/components/jobPosting/JobPostingStages.js]: Name is probably a better key than index especially if name is unique

File: client/src/components/jobPosting/JobPostingStage.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { Chip } from '@material-ui/core'
```
Comment (llmlks) [inline client/src/components/jobPosting/JobPostingStage.js]: Andrea found [information about how to minimize bundle size](https://material-ui.com/guides/minimizing-bundle-size/)

File: client/src/components/jobPosting/JobPostingStages.js (hunk 5 above)
Comment (llmlks) [reply client/src/components/jobPosting/JobPostingStages.js]: This comment is still valid

**Pull request merged**